### ISSUE TITLE
All-in-one bug-fix addon

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#7152]
### Why this would be helpful

Sometimes there's a bug or undesired behavior in Scratch that we want to fix with an addon. If I got them all, these include:
- Faster project loading
- Resizable comment input?
- Place backpack code at mouse
- Better editor comments
- Do not automatically run duplicated blocks
- Fix sprite pane glitching
- Redirect to standard forums (now that the old mobile view of the forums has been taken down)
- Remember collapsed forum categories
- SVG uploads fix
- The `Use previous color instead of resetting after switching tools` setting in Customizable default costume editor colors

The main issue with these is that the user has to be looking out for them when they're a great opportunity to have bug fixes enabled by default. Surprisingly, most of these are not enabled by default.

Some of these are also so minor they almost don't warrant a whole addon.

### How this addon works

This addon would just be a collection of bug fixes and other subconscious UX enhancements that objectively, everyone would use. It'd be enabled by default and each bug fix would be a setting, so you could see what is being patched, and turn any patches off if for some reason you want to.

Some of the above addons could be moved into this addon.

Perhaps this addon could have a confirmation when being disabled, so no one accidentally turns it off and ruins parts of their experience.

### Possible alternatives

We could also add a "Bug fixes" category of addons and separate these fixes into their own addons which would only show up in this category.

### Additional context

Related:
- #6409

**Comment by Samq64** [comment]
Related:
- #2562
- #2617

**Comment by Secret-chest** [comment]
I don't like the monolithic addons thing. I suggest making a bugfix group like Forums and enabling them all by default.

**Comment by WorldLanguages** [comment]
In theory this could be a single addon, but we would need setting descriptions. Setting names wouldn't be enough to explain each bug that is fixed.

**Comment by DNin01** [comment]
> we would need setting descriptions.

#5320

**Comment by DNin01** [comment]
I also kind of like the sound of separating each bug fix into its own addon and putting them all into a "Bug fixes" group, like @Secret-chest said, with each addon being enabled by default. To reduce clutter, they would not show up in the "Enabled" group but would be searchable. That way, you can search and see if we have a fix for something and then find out we're already fixing it for you.

![Group of bug fixes in a list of addons, all of which are enabled](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/53fb3b01-8028-46df-832d-0f6023ce9e70)

Micro-addons!

**Comment by BroJac5246** [comment]
> I also kind of like the sound of separating each bug fix into its own addon and putting them all into a "Bug fixes" group, like @Secret-chest said, with each addon being enabled by default. To reduce clutter, they would not show up in the "Enabled" group but would be searchable. That way, you can search and see if we have a fix for something and then find out we're already fixing it for you.
> 
> ![Group of bug fixes in a list of addons, all of which are enabled](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/53fb3b01-8028-46df-832d-0f6023ce9e70)
> 
> Micro-addons!

I've been thinking about a model like this for a 2.0 version of the extension, where addons that don't add stuff but rather make small- to medium-sized changes are called "tweaks." This would fit in very well with that.

**Comment by DNin01** [comment]
I'm currently working on adding a bug fixes group, storing the last version the extension so we can tell if the user is updating to a new version, enabling some of these addons by default now, and coding a gradual rollout mechanism for the `fix-uploaded-svgs` addon (just in case there's a bug we haven't caught).

**Comment by DNin01** [comment]
Once we implement a bug fixes or minor addons group, I'd like to make two addons:
- Make the backdrop behind the add costume/sound button click-through
- Allow the backpack to be scrolled without having to hold Shift (which might be browser-specific).

**Comment by Secret-chest** [comment]
> I've been thinking about a model like this for a 2.0 version of the extension, where addons that don't add stuff but rather make small- to medium-sized changes are called "tweaks." This would fit in very well with that.

It could be a group, but not more than that.

**Comment by BroJac5246** [comment]
> > I've been thinking about a model like this for a 2.0 version of the extension, where addons that don't add stuff but rather make small- to medium-sized changes are called "tweaks." This would fit in very well with that.
> 
> It could be a group, but not more than that.

Yeah that's what it would pretty much be. It probably wouldn't be too hard to implement.
